### PR TITLE
[trust] Basic plumbing for Monberaux

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

**Comment by zach2good** [comment]
Going to leave this in its current state, someone else can finish this off/polish it.

**Comment by zach2good** [comment]
Heavily stripped this down and left a lot of comments in the code about whats missing and what needs to be done in the future, ready for review (and eventual merge), as this doesn't add anything or make him playable

**Review by TeoTwawki — APPROVED** [review]

File: scripts/globals/mobskills/mix_panacea-1.lua [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+require("scripts/globals/status")
+require("scripts/globals/mobskills")
+-----------------------------------
+local mobskillObject = {}
+
+mobskillObject.onMobSkillCheck = function(target, mob, skill)
+    return 0
+end
+
+mobskillObject.onMobWeaponSkill = function(target, mob, skill)
```
Comment (claywar) [inline scripts/globals/mobskills/mix_panacea-1.lua:45]: This looks like it should be tabled and iterated over

File: scripts/globals/mobskills/holy_water.lua [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+require("scripts/globals/status")
+require("scripts/globals/mobskills")
+-----------------------------------
+local mobskillObject = {}
+
+mobskillObject.onMobSkillCheck = function(target, mob, skill)
+    return 0
+end
+
+mobskillObject.onMobWeaponSkill = function(target, mob, skill)
```
Comment (claywar) [inline scripts/globals/mobskills/holy_water.lua:21]: Would prefer a table here, but it's relatively small

**Review by claywar — APPROVED** [review]
Seems reasonable. Future CI will not like some things, but I can take care of those. Also one unused that luacheck called out.

**Review by jmcmorris — COMMENTED** [review]
What if we made a helper function that takes an entity and a list of status effects. It then can iterate over those status effects checking if the entity has it and if so then removes it.

**Comment by zach2good** [comment]
> checking if the entity has it and if so then removes it.

I've updated the PR, there's little need to do this: delStatusEffect is implemented as a range-for-loop looking for a specific id, so it can delete it. So there's no need for a check

**Comment by zach2good** [comment]
Also, `hasStatusEffect` is also implemented as a loop, so checking doesn't save any time over the delete

**Comment by TeoTwawki** [comment]
> > checking if the entity has it and if so then removes it.
> 
> I've updated the PR, there's little need to do this: delStatusEffect is implemented as a range-for-loop looking for a specific id, so it can delete it. So there's no need for a check

forgetting the check if it has part of that for a minute, I think in the future it may still be useful to be able to feed a table to a function (or bake this into the existing binding even) to do multiple status as a table without typing out the same loop every time we table a bunch of effects in lua land - I've seen us use a table and for loop to del multiple effects several times when a flag didn't cover all the effects (some just check erasable/dispellable)

**Pull request merged**